python/packages/ai/teams/ai/models/openai/planner.py

```diff
@@ -115,10 +137,85 @@ async def generate_plan(
                         new_commands.append(command)
                 plan.commands = new_commands
 
-            return plan
+            return await self.review_plan(
+                turn_context,
+                state,
+                plan,
+            )
 
         return Plan()
 
+    async def review_prompt(
+        self,
+        _context: TurnContext,
+        state: TurnState,
+        _prompt: PromptTemplate,
+    ) -> Optional[Plan]:
+        if self._options.moderate == "output":
+            return None
+
+        try:
+            res = await self._client.create_moderation(
+                state.temp.input,
+            )
+
+            if res.data.results[0].flagged:
+                return Plan(
+                    commands=[
+                        PredictedDoCommand(
+                            action=ActionTypes.FLAGGED_INPUT,
+                            entities=res.data.results[0].__dict__,
```

Comment:
minor: `vars()` seems more preferred than `__dict__`, though they're doing the same thing. https://stackoverflow.com/questions/21297203/use-dict-or-vars